### DISCUSSION TITLE
JWT 토큰 사용 관련 논의

### DISCUSSION THREAD

**Discussion by Dae-Hwa** [Sc0up/scoup-backend#80] [New Feature]
- Access Token과 Refresh Token 사용
  - Access Token 보관 장소 협의 필요
  - Refresh Token은 cookie http only로 사용 예정

- Refresh Token 관리를 위한 스토리지
  - db -> redis로 갈지 처음부터 redis로 갈지 논의

- Refresh Token 관리 정책
  - 화이트 리스트 vs 블랙 리스트
  - token만 저장 vs ip user agent 등으로 보안 강화
    - 보안유지가 꼭 필요한지도 논의 필요

**Comment by janeljs** [comment]
프레디가 디스커션을 적극 활용하고 계셨네요ㅋㅋㅋㅋ 이제 확인해서 죄송해요! 
저번에 잠깐 논의하기는 했지만 개인적인 생각으로는 
1. access token -> 굳이 보관할 필요가 있을까요? 아직은 소셜 서비스와 연동되는 기능이 회원가입 및 로그인밖에 없어서 필요 없는 거 같아요! 나중에 필요하게 되면 userRepository에 token 필드 추가해서 저장하는 건 어떤지..? 그리고 Refresh Token 쿠키에서 관리하는 거 좋은 것 같습니당
2. 음 처음부터 redis로 간다!ㅋㅋㅋ 재밌는 건 빨리 해보는게 좋지 않을까요.
3. 저는 화이트 리스트로 해서 사전에 인증된 토큰만 정상인 것으로 확인하는게 좋을 것 같아요! user agent의 경우는 사용자가 기기를 바꿀 시에 다시 로그인 해야하게 하고 싶어서 그런건가요?

**Comment by Dae-Hwa** [comment]
JWT는 근본적으로 취약하다. 그러면 추가 보안이 필요할 수 있다.
추가 로그인이 필요한 경우?

**Reply by Dae-Hwa** [reply]
콜백 url을 같이 넘긴 뒤 전송?
